src/BufferedRequest.ts

```diff
@@ -0,0 +1,38 @@
+import { Headers } from 'headers-polyfill/lib'
+import { decodeBuf } from './utils/bufferCodec'
+import { uuidv4 } from './utils/uuid'
+
+export class BufferedRequest {
+  public id = uuidv4()
+
+  constructor(
+    public readonly url: URL,
+    private readonly body: ArrayBuffer,
+    private readonly init: RequestInit
```

Comment:
Unfortunately, `RequestInit` is only defined in `lib/dom` in TypeScript. It will not exist in Node.js.

We need to re-defined the request init manually:

```ts
export interface RequestInit {
  method: string
  headers: Record<string, string | string[]> | Headers
  credentials: RequestCredentials // our internal type, can reuse
  body: ArrayBuffer
}
```

Let's also change the constructor signature to match that of the Fetch API Request:

```ts
new Request(url)
new Request(url, { headers: {}, body: new ArrayBuffer() })
```

Reply:
Types are stripped at compile time. So, this is not required at runtime by nodejs. 

I'll change the constructor signature to be more fetch compatible.

Reply:
Types are stripped from the code but not from the type definitions. `BufferedRequest.d.ts` will reference global scope `Request` and will cause a type violation when people will run it Node in TypeScript projects.